text/0046-metadata-for-offline-signers.md

```diff
@@ -85,17 +97,40 @@ enum MetadataDescriptor {
 }
 
 struct MetadataDescriptorV1 {
-  extrinsic_metadata: ExtrinsicMetadata
-  spec_version: String
-  spec_name: String
+  call_ty: u32, // TypeId
+  signed_extensions: Vec<SignedExtensionsMetadata>,
+  spec_version: String,
+  spec_name: String,
   ss58_prefix: u16,
   decimals: u8,
   token_symbol: String
 }
+
+struct SignedExtensionMetadata {
+  identifier: String,
+  ty: u32, // TypeId
+  additional_signed: u32, // TypeId
+}
 ```
 
 constitute metadata descriptor. This is minimal information that is, together with (shortened) types registry, sufficient to decode any signable transaction.
 
+**Note on `spec_version`**: this type is described in metadata; it is indeed u32 in Polkadot and in Kusama and most other networks and thus could be defined in typesafe manner, but theoretically it could be anything, probably even enum similar to how Era is constructed - why not? - as long as it is serializable. Something like 
+
+```
+struct version {
+  major: u32,
+  minor: u32,
+  notes: String,
+}
+```
+
+would be reasonable solution that might be adopted in future.
+
+We've really seen a couple of networks using u16 there in the past. Not that it's all that important, but as long as our protocol would be the only limitation - it would be a limitation for our protocol; it would be unwise to enforce it here without first enforcing it upstream.
+
+The version is not exactly algebraic number - it has comparison operation, but no other algebra is defined; the primitive variable with these properties is really `String`, not integer. This is silly, we've argued a lot about this internally, but with naive JS-style solution we have all the flexibility we might want and no real downsides (as from point of view of our protocol, it's just an opaque constant - key for database search).
```

Comment:
```suggestion
We saw a couple of networks using u16 there in the past. Not that it's all that important, but as long as our protocol would be the only limitation - it would be a limitation for our protocol; it would be unwise to enforce it here without first enforcing it upstream.

The version is not exactly an algebraic number - it has comparison operation, but no other algebra is defined; the primitive variable with these properties is really a `String`, not an integer. We've argued a lot about this internally, but with a naive JS-style solution, we have all the flexibility we might want and no real downsides (as from the point of view of our protocol, it's just an opaque constant - key for database search).
```